cli/cmd/update_project.go

```diff
@@ -0,0 +1,121 @@
+package cmd
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/keptn/keptn/cli/pkg/websockethelper"
+
+	apimodels "github.com/keptn/go-utils/pkg/api/models"
+	apiutils "github.com/keptn/go-utils/pkg/api/utils"
+	keptn "github.com/keptn/go-utils/pkg/lib"
+	"github.com/keptn/keptn/cli/pkg/credentialmanager"
+	"github.com/keptn/keptn/cli/pkg/logging"
+	"github.com/spf13/cobra"
+)
+
+type updateProjectCmdParams struct {
+	GitUser   *string
+	GitToken  *string
+	RemoteURL *string
+}
+
+var updateProjectParams *updateProjectCmdParams
+
+// upProjectCmd represents the project command
+var upProjectCmd = &cobra.Command{
+	Use:   "project PROJECTNAME --git-user=GIT_USER --git-token=GIT_TOKEN --git-remote-url=GIT_REMOTE_URL",
+	Short: "Updates an existing keptn project",
+	Long: `Updates an existing keptn project with the provided name. 
+Updating a shipyard file is not possible.
+
+By executing the *update project* command, keptn will add the provided upstream repository to the existing internal Git repository that is used to maintain all project-related resources. 
+To upstream this internal Git repository to a remote repository, the Git user (*--git-user*), an access token (*--git-token*), and the remote URL (*--git-remote-url*) are required.
+
+For more information about updateing projects or upstream repositories visit https://keptn.sh/docs/develop/manage/project/ .
+`,
+	Example:      `keptn update project PROJECTNAME --git-user=GIT_USER --git-token=GIT_TOKEN --git-remote-url=GIT_REMOTE_URL`,
+	SilenceUsage: true,
+	Args: func(cmd *cobra.Command, args []string) error {
+		_, _, err := credentialmanager.NewCredentialManager().GetCreds()
+		if err != nil {
+			return errors.New(authErrorMsg)
+		}
+
+		if len(args) != 1 {
+			cmd.SilenceUsage = false
+			return errors.New("required argument PROJECTNAME not set")
+		}
+
+		if !keptn.ValidateKeptnEntityName(args[0]) {
+			errorMsg := "Project name contains upper case letter(s) or special character(s).\n"
+			errorMsg += "Keptn relies on the following conventions: "
+			errorMsg += "start with a lower case letter, then lower case letters, numbers, and hyphens are allowed.\n"
+			errorMsg += "Please update project name and try again."
+			return errors.New(errorMsg)
+		}
+		return nil
+	},
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+
+		if *updateProjectParams.GitUser == "" && *updateProjectParams.GitToken == "" && *updateProjectParams.RemoteURL == "" {
```

Comment:
Is there a reason for this check? I think that a project update without those parameters does not make sense.